src/mlpack/tests/activation_functions_test.cpp

```diff
@@ -325,6 +327,77 @@ void CheckPReLUGradientCorrect(const arma::colvec input,
   BOOST_REQUIRE_CLOSE(gradient(0), target(0), 1e-3);
 }
 
+/*
+ * Simple SELU activation test to check whether the mean and variance remain
+ * invariant after passing normalized inputs through the function.
+ */
+BOOST_AUTO_TEST_CASE(SELUFunctionNormalizedTest)
+{
+  arma::mat input = arma::randn<arma::mat>(1000, 1);
+
+  arma::mat output;
+
+  SELU selu;
+
+  selu.Forward(std::move(input), output);
+
+  BOOST_REQUIRE_LE(arma::as_scalar(arma::abs(arma::mean(input) -
+      arma::mean(output))), 0.1);
+
+  BOOST_REQUIRE_LE(arma::as_scalar(arma::abs(arma::var(input) -
+      arma::var(output))), 0.1);
+}
+
+/*
+ * Simple SELU activation test to check whether the mean and variance
+ * vary significantly after passing unnormalized inputs through the function.
+ */
+BOOST_AUTO_TEST_CASE(SELUFunctionUnnormalizedTest)
+{
+  const arma::colvec input("5.96402758 0.9966824 0.99975321 1 \
+                            7.76159416 -0.76159416 0.96402758 8");
+
+  arma::mat output;
+
+  SELU selu;
+
+  selu.Forward(std::move(input), output);
+
+  BOOST_REQUIRE_GE(arma::as_scalar(arma::abs(arma::mean(input) -
+      arma::mean(output))), 0.1);
+
+  BOOST_REQUIRE_GE(arma::as_scalar(arma::abs(arma::var(input) -
+      arma::var(output))), 0.1);
+}
+
+/*
+ * Simple SELU derivative test to check whether the derivatives
+ * produced by the activation function are correct.
+ *
+ */
+BOOST_AUTO_TEST_CASE(SELUFunctionDerivativeTest)
+{
+  arma::mat input = arma::ones<arma::mat>(1000, 1);
+
+  arma::mat error = arma::ones<arma::mat>(input.n_elem, 1);
+
+  arma::mat derivatives;
+
+  SELU selu;
+
+  selu.Backward(std::move(input), std::move(error), std::move(derivatives));
+
+  BOOST_REQUIRE_LE(arma::as_scalar(arma::abs(arma::mean(derivatives) -
+      selu.Lambda())), 10e-5);
+
+  input.fill(-1);
+
+  selu.Backward(std::move(input), std::move(error), std::move(derivatives));
+
+  BOOST_REQUIRE_LE(arma::as_scalar(arma::abs(arma::mean(derivatives) -
+      selu.Lambda()*(selu.Alpha()-1))), 10e-5);
```

Comment:
There are some horizontal whitespace issues here; if you can fix that it would be great: `selu.Lambda() * (selu.Alpha() - 1)`.  Also it seems to me like it would make more sense to use `1e-4` throughout instead of `10e-5`.